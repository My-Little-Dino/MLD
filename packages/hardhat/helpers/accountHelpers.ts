import { fromBech32 } from '@harmony-js/crypto';
import { isBech32Address } from '@harmony-js/utils';

export const parseOneAddress = (address: string): string => (isBech32Address(address) ? fromBech32(address) : address);
