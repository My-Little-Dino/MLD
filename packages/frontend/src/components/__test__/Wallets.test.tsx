import React from 'react';
import ReactDOM from 'react-dom';
import { fireEvent, render } from '@testing-library/react';

import { useWeb3React } from '@web3-react/core';
import Wallets, { Props } from '../Wallets';

import { connectorsByName } from '../../utils/connectors';
import { mapWallets } from '../../helpers/walletHelpers';

jest.mock('@web3-react/core');

describe('Test Wallets component', () => {
	const testProps: Props = {
		closeModal: jest.fn(),
	};
	const component = <Wallets {...testProps} />;

	const setup = () => render(component);

	beforeEach(() => {
		(useWeb3React as jest.Mock).mockReturnValue({
			activate: jest.fn(),
		});
	});

	afterEach(jest.clearAllMocks);

	it('Should render component without crash', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Should match snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});

	it('Should call activate this wallet & close modal when click a wallet', () => {
		const walletName = 'OneWallet';
		const { getByText } = setup();

		const oneWallet = getByText(mapWallets[walletName].name);
		fireEvent.click(oneWallet);

		expect(useWeb3React().activate).toHaveBeenCalledWith(connectorsByName[walletName]);
		expect(testProps.closeModal).toHaveBeenCalled();
	});
});
