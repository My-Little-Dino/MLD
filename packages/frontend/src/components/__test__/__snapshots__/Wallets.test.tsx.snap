// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Wallets component Should match snapshot 1`] = `
<div>
  <div
    class="sc-gtsrHT dURUhH"
  >
    <div
      class="sc-dlnjwi GMoHn"
    >
      <img
        class="sc-bdnxRM kIwPUV"
        src="one.svg"
      />
      One wallet
    </div>
    <div
      class="sc-dlnjwi GMoHn"
    >
      <img
        class="sc-bdnxRM kIwPUV"
        src="math.png"
      />
      Math wallet
    </div>
    <div
      class="sc-dlnjwi GMoHn"
    >
      <img
        class="sc-bdnxRM kIwPUV"
        src="metamask.svg"
      />
      Metamask
    </div>
  </div>
</div>
`;
